docs/ref/models/querysets.txt

```diff
@@ -1925,19 +1937,14 @@ when you're using manually specified primary keys. If an object needs to be
 created and the key already exists in the database, an
 :exc:`~django.db.IntegrityError` will be raised.
 
-This method is atomic assuming correct usage, correct database configuration,
-and correct behavior of the underlying database. However, if uniqueness is not
-enforced at the database level for the ``kwargs`` used in a ``get_or_create``
-call (see :attr:`~django.db.models.Field.unique` or
-:attr:`~django.db.models.Options.unique_together`), this method is prone to a
-race-condition which can result in multiple rows with the same parameters being
-inserted simultaneously.
-
-If you are using MySQL, be sure to use the ``READ COMMITTED`` isolation level
-rather than ``REPEATABLE READ`` (the default), otherwise you may see cases
-where ``get_or_create`` will raise an :exc:`~django.db.IntegrityError` but the
-object won't appear in a subsequent :meth:`~django.db.models.query.QuerySet.get`
-call.
+If you are using MySQL with the default ``REPEATABLE READ``
```

Comment:
Tim commented on the issue that since https://github.com/django/django/commit/924af638e4d4fb8eb46a19ac0cafcb2e83480cf3 Django defaults to `READ COMMITTED`, so, at least, the "default" is ambiguous here. (Sure it's MySQL's default, but likely you won't see it.) 

I'm thinking this discussion is better moved (almost entirely) to `docs/ref/databases.txt` at this point, in the `mysql-isolation-level` section. Thoughts on that?

Reply:
I don't think this paragraph has much value now, since users changing their isolation level away from Django's default should be aware of the consequences at a deeper lower level than "get_or_create might not work", and they should understand that `IntegrityError` will mean the row isn't written. But it's probably worth keeping for the sake of completeness and the database docs are a more appropriate place—anyone changing the isolation level should probably be reading that section of the docs before doing so. And yes we can just drop the word "default" due to its ambiguity.

Reply:
I deleted this paragraph and added a brief clause to the "Isolation level" section of the MySQL database documentation which links back to the `get_or_create` documentation.